src/http.cc

```diff
@@ -1393,6 +1395,16 @@ HttpStateData::truncateVirginBody()
     }
 }
 
+/// called when an incomplete reply body preliminary ends with unexpected EOF
+void
+HttpStateData::failOnPrematureReplyBodyEof()
+{
+    const auto err = new ErrorState(ERR_READ_ERROR, Http::scBadGateway, fwd->request, fwd->al);
```

Comment:
Just a note FTR (no changes requested here): We have discussed that creating ErrorState so late in the game (i.e. when there is a half-filled entry already) is wasteful (at least) and possibly dangerous. I found that we already have an error type that is not as "heavy" as ErrorState! It is called `Error` (declared in src/error/Error.h). At some point, FwdState::err may need to be downgraded to that simpler type, but I hope that we can continue to reuse the existing ErrorState `err` for this PR.